client/components/theme/index.jsx

```diff
@@ -314,23 +316,24 @@ export class Theme extends Component {
 	};
 
 	renderBadge = () => {
-		const isLockedStyleVariation =
-			this.props.shouldLimitGlobalStyles &&
-			! isDefaultGlobalStylesVariationSlug( this.props.selectedStyleVariation?.slug );
+		const { selectedStyleVariation, shouldLimitGlobalStyles, siteId, siteSlug, theme } = this.props;
+
+		const premiumPlanSlug = getPlan( PLAN_PREMIUM ).getPathSlug();
+		const isLocked = isLockedStyleVariation( {
+			isPremiumTheme: theme.theme_tier?.slug === premiumPlanSlug,
```

Comment:
Since theme tiers are still under development, should we use a feature flag here? Which leads me to my next question: How can we know if the current theme is premium without the feature fag active?

Reply:
> How can we know if the current theme is premium without the feature fag active?

If I disable the flag, I still get the `theme_tier` data. Isn't that expected?

Reply:
It's expected, but I don't think you should trust the data it if the flag is disabled since right now we're internally testing the theme tiers and maybe there are themes assigned to the premium tier due to testing purposes which shouldn't affect production.

Reply:
I understand. So, if there's no way to know whether a theme is premium when the flag is disabled, I'd just check the premium theme value when the flag is enabled, like:

```
export function isLockedStyleVariation( {
	isPremiumTheme,
	styleVariationSlug,
	shouldLimitGlobalStyles,
}: {
	isPremiumTheme: boolean;
	styleVariationSlug?: StyleVariation[ 'slug' ];
	shouldLimitGlobalStyles?: boolean;
} ) {
	return (
		( shouldLimitGlobalStyles && isEnabled( 'themes/tiers' )
			? ! isPremiumTheme
			: true && ! isDefaultGlobalStylesVariationSlug( styleVariationSlug ) ) ?? false
	);
}
```

Wdyt?

Reply:
Ah, cool! Added on https://github.com/Automattic/wp-calypso/pull/86654/commits/4577745824882a2d33116da8221e93e876247288